frontend/src/concepts/trustyai/content/useTrustyExistingSecretValidation.ts

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { UseTrustyInstallModalDataExisting } from '~/concepts/trustyai/content/useTrustyInstallModalData';
+
+type UseTrustyExistingSecretValidation = {
+  onBlur: () => void;
+};
+
+const useTrustyExistingSecretValidation = (
+  formData: UseTrustyInstallModalDataExisting,
+): UseTrustyExistingSecretValidation => {
+  const { data, onCheckState } = formData;
+
+  const timeoutRef = React.useRef<NodeJS.Timeout | null>(null);
+
+  const clearCheckTimeout = React.useCallback(() => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (data) {
+      timeoutRef.current = setTimeout(onCheckState, 3000);
+    }
+
+    return clearCheckTimeout;
+  }, [clearCheckTimeout, data, onCheckState]);
+
```

Comment:
why not use our debounce hook?

Reply:
I think I looked for it 🤔 Maybe I thought I did and then didn't lol...

Reply:
TODO for me